openshift/templates/monitoring/grafana-template.yml

```diff
@@ -0,0 +1,219 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: pims-grafana-deploy
+  annotations:
+    openshift.io/display-name: PIMS Grafana
+    description: Deployment template for Grafana
+    tags: pims-monitoring
+objects:
+  - kind: ConfigMap
+    apiVersion: v1
+    metadata:
+      name: pims-monitoring-tools-grafana-datasource-config
+      namespace: jcxjin-tools
+      annotations:
+        description: Grafana datasource configuration
+      labels:
+        name: pims-monitoring-tools-grafana-datasource-config
+        app: pims
+        component: monitoring
+        env: tools
+        role: prometheus
+    type: Opaque
+    data:
+      datasources.yml: |-
+        apiVersion: 1
+        datasources:
+        - name: Prometheus
+          type: prometheus
+          access: proxy
+          url: https://prometheus-jcxjin-tools.pathfinder.gov.bc.ca
+
+  - kind: ConfigMap
+    apiVersion: v1
+    metadata:
+      name: pims-monitoring-tools-grafana-ini-config
+      namespace: jcxjin-tools
+      annotations:
+        description: Grafana ini configuration
+      labels:
+        name: pims-monitoring-tools-grafana-ini-config
+        app: pims
+        component: monitoring
+        env: tools
+        role: prometheus
+    type: Opaque
+    data:
+      grafana.ini: |-
+        [server]
+        root_url = https://grafana-jcxjin-tools.pathfinder.gov.bc.ca/
+        domain = grafana-jcxjin-tools.pathfinder.gov.bc.ca
+        enforce_domain = true
+        [auth.github]
+        enabled = true
+        allow_sign_up = true
+        client_id = ${GITHUB_CLIENT_ID}
+        client_secret = ${GITHUB_CLIENT_SECRET}
+        scopes = user:email,read:org
+        allowed_organizations = ${GITHUB_ALLOWED_ORG}
+        auth_url = https://github.com/login/oauth/authorize
+        token_url = https://github.com/login/oauth/access_token
+        api_url = https://api.github.com/user
+
+  - kind: ServiceAccount
+    apiVersion: v1
+    metadata:
+      name: ${GRAFANA_SERVICE_ACCOUNT_NAME}
+      labels:
+        name: ${GRAFANA_SERVICE_ACCOUNT_NAME}
+      namespace: jcxjin-tools
+  - kind: Route
+    apiVersion: v1
+    metadata:
+      labels:
+        name: grafana
+      name: grafana
+      namespace: jcxjin-tools
+    spec:
+      host: "grafana-jcxjin-tools.pathfinder.gov.bc.ca"
```

Comment:
Should probably be a variable
I wonder if we should rename to something more generic `pims-grafana.pathfinder.gov.bc.ca`?